substrate/frame/revive/uapi/src/host/riscv64.rs

```diff
@@ -59,9 +59,30 @@ mod sys {
 			out_ptr: *mut u8,
 			out_len_ptr: *mut u32,
 		) -> ReturnCode;
-		pub fn call(ptr: *const u8) -> ReturnCode;
-		pub fn delegate_call(ptr: *const u8) -> ReturnCode;
-		pub fn instantiate(ptr: *const u8) -> ReturnCode;
+		pub fn call(
+			flags_and_callee: u64,
+			ref_time_limit: u64,
+			proof_size_limit: u64,
+			deposit_and_value: u64,
+			input_data: u64,
+			output_data: u64,
+		) -> ReturnCode;
+		pub fn delegate_call(
+			flags_and_callee: u64,
+			ref_time_limit: u64,
+			proof_size_limit: u64,
+			deposit_ptr: *const u32,
```

Comment:
Should just point to bytes?

```suggestion
			deposit_ptr: *const u8,
```

Reply:
Good catch thanks!